sdk/python/test/sdk_unittest/test_training_client.py

```diff
@@ -0,0 +1,193 @@
+import multiprocessing
+import unittest
+from unittest.mock import patch, Mock
```

Comment:
Would it be better to use `pytest` for our SDK unit tests ?
As @droctothorpe mentioned here: https://github.com/kubeflow/katib/issues/2184#issuecomment-1660475384, we want to unify test framework across all Python files.

We already using pytest for our e2e tests: https://github.com/kubeflow/training-operator/blob/master/sdk/python/test/e2e/test_e2e_mpijob.py#L17.

WDYT @deepanker13 @johnugeorge @tenzen-y @droctothorpe ?

Reply:
+1 on unifying around pytest.

Reply:
+1 on pytest.

Reply:
All the test_kubeflow_org_v1* files generated by openapi-generator are using unittest, that's why I used the same. Also pytest supports running Python unittest -based tests

Reply:
I understand this @deepanker13, but I think we should minimize number of dependencies that we have for our SDK.
E.g. it is better to remove `"parameterized>=0.9.0"` from the SDK requirements if we can use `pytest` to parameterize our test cases like here: https://github.com/kubeflow/katib/blob/master/test/unit/v1beta1/suggestion/test_optuna_service.py#L35-L44

Also, it is important to keep our tests consistent across various Kubeflow components (e.g. Pipelines, Katib, Training Operator). In that case, it will be much easier to onboard a new contributors to the project.
That is why I suggested to use `pytest` and try to keep tests structure consistent.

What do you think ?

Reply:
Sure, I have updated the code